[Bug]: Auto Format and Lint no longer working

### Summary Of Actions

When a PR is submitted and it hasn't been linted the test fails but only after trying to make the changes and push them to the PR.

### Operating System

Macos

### What Python Version are you using?

Python 3.12

### Have you installed Plugins/Themes/Custom Parsers?

N/A

### What did you enter to get the output that you received?

```Shell
This was the response based on a PR.
```


### Output

```Shell
Run if [ -n "$(git status --porcelain)" ]; then
  if [ -n "$(git status --porcelain)" ]; then
    git config --global user.name "Ruff"
    git config --global user.email "ruff-action@users.noreply.github.com"
    git add -A
    git commit -m "Format and lint with Ruff"
    git push origin HEAD:caching-workflows
  fi
...
remote: Permission to render-engine/render-engine.git denied to github-actions[bot].
fatal: unable to access 'https://github.com/render-engine/render-engine/': The requested URL returned error: 403
Error: Process completed with exit code 12[8](https://github.com/render-engine/render-engine/actions/runs/11504893660/job/32100071862?pr=831#step:9:8).
```
```

--- a/.github/workflows/lint.yml
+++ b/.github/workflows/lint.yml
@@ -1,13 +1,16 @@
-name: Format and Lint with Ruff and MarkdownLint
+name: Format and Lint
 on:
-    pull_request:
-        branches: [main]
-    workflow_dispatch:
+  pull_request:
+    branches: [main]
+  workflow_dispatch:
+
 permissions:
-    contents: write
-    pull-requests: write
+  contents: write
+  pull-requests: write
+
 jobs:
-  build:
+  lint-python:
+    name: Format and Lint Python
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
@@ -21,33 +24,25 @@ jobs:
         run: |
           python -m pip install --upgrade pip
           pip install ruff
-      # Update output format to enable automatic inline annotations.
-      - name: Run Ruff
-        run: |
-            ruff check . --fix --output-format=github
-            ruff format .
+      - name: Check Linting
+        run: ruff check . --output-format=github
+      - name: Check formatting
+        run: ruff format . --check
 
+  lint-docs:
+    name: Lint Markdown
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
       - name: Markdown Lint base-files
         uses: DavidAnson/markdownlint-cli2-action@v18
         with:
-            fix: true
-            globs: |
-              *.md
-              .github/**/*.md
-        continue-on-error: true
+          globs: |
+            *.md
+            .github/**/*.md
+          config: ./.markdownlint.json
       - name: Markdown Lint Docs
         uses: DavidAnson/markdownlint-cli2-action@v18
         with:
-            fix: true
-            globs: docs/**/*.md
-            config: docs/.markdownlint.json
-        continue-on-error: true
-      - name: Commit changes
-        run: |
-          if [ -n "$(git status --porcelain)" ]; then
-            git config --global user.name "Ruff"
-            git config --global user.email "ruff-action@users.noreply.github.com"
-            git add -A
-            git commit -m "Format and lint with Ruff"
-            git push origin HEAD:${{ github.event.pull_request.head.ref }}
-          fi
+          globs: docs/**/*.md
+          config: docs/.markdownlint.json

--- a/README.md
+++ b/README.md
@@ -1,6 +1,6 @@
 <!-- ALL-CONTRIBUTORS-BADGE:START - Do not remove or modify this section -->
 
-[![All Contributors](https://img.shields.io/badge/all_contributors-17-orange.svg?style=flat-square)](#contributors-)
+[![All Contributors](https://img.shields.io/badge/all_contributors-17-orange.svg?style=flat-square)](#contributors)
 
 <!-- ALL-CONTRIBUTORS-BADGE:END -->
 

